### PR TITLE
feat: utilize System.CommandLine to add Help args

### LINKED ISSUE #3: Add help (-h/--help) argument
To track when this particular issue is resolved.

> # Add help (-h/--help) argument
>
> This should just show the same help message shown when NupkgDeterministicator is ran with no args.
>
> An argument like this, which always exits with code 0, would be helpful in cross-platform MSBuild tasks to check if NupkgDeterministicator is installed before executing it after `Pack`.
> 
> ```
> NupkgDeterministicator.exe -h
> File -h doesn't exist
> Unhandled exception. System.IO.FileNotFoundException: Could not find file 'C:\Repos\HaloSPV3\hce.shared-config\-h'.
> File name: 'C:\Repos\HaloSPV3\hce.shared-config\-h'
>    at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)     
>    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
>    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
>    at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
>    at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
>    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean useAsync)
>    at System.IO.Compression.ZipFile.Open(String archiveFileName, ZipArchiveMode mode, Encoding entryNameEncoding)
>    at System.IO.Compression.ZipFile.ExtractToDirectory(String sourceArchiveFileName, String destinationDirectoryName, Encoding entryNameEncoding, Boolean overwriteFiles)
>    at Kuinox.NupkgDeterministicator.Build.UnzipPackage(String package, String dest) in C:\dev\Kuinox.NupkgDeterministicator\Kuinox.NupkgDeterministicator\Program.cs:line 76
>    at Kuinox.NupkgDeterministicator.Build.RepackNugetPackage(String path, DateTime dateTime) in C:\dev\Kuinox.NupkgDeterministicator\Kuinox.NupkgDeterministicator\Program.cs:line 58
>    at Kuinox.NupkgDeterministicator.Build.Main(String[] args) in C:\dev\Kuinox.NupkgDeterministicator\Kuinox.NupkgDeterministicator\Program.cs:line 38
> ```

### PR DESCRIPTION
**Pull request by BinToss**
Will resolve #3

Note: this Pull Request aims to maintain Public API and CLI compatibility. This *shouldn't* break anything.

### CODE REVIEW

**Comment by BinToss** [comment]
https://asciinema.org/a/kup9OfP80TnQwsbEziBtFTbE5
powersession-playback

File: Kuinox.NupkgDeterministicator/Program.cs [hunk 1]
```diff
@@ -18,30 +19,51 @@ public class Build
 
     public static int Main(string[] args)
     {
-        if (args.Length == 0)
+        // todo: a directory path should imply all package files (*.nupkg, *.snupkg) in the directory should be determinized
```
Comment (Kuinox) [inline Kuinox.NupkgDeterministicator/Program.cs]: Why not keeping a single path as input and have the caller do the job of iterating instead ?

File: Kuinox.NupkgDeterministicator/Program.cs [hunk 2]
```diff
@@ -18,30 +19,51 @@ public class Build
...
+            argPathToNupkg,
+            argDateTime
+        );
+
+        return rootCommand.Invoke(args);
     }
 
+    static readonly DateTime DefaultDateTime = new(2000, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
     const string Help =
```
Comment (Kuinox) [inline Kuinox.NupkgDeterministicator/Program.cs]: You can inline it now that's it's used once.

File: Kuinox.NupkgDeterministicator/Program.cs (hunk 1 above)
Comment (BinToss) [reply Kuinox.NupkgDeterministicator/Program.cs]: I had forgotten MSBuild has [batching](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-batching?view=vs-2022) for iteration.
I'll remove this todo and the `--datetime` todo.

**Comment by Kuinox** [comment]
I will manually test everything with the msbuild PR

**Pull request merged**